nautobot/docs/development/core/ui-component-framework.md

```diff
@@ -0,0 +1,408 @@
+                             # Will count all devices related to this object
+        
+        (Circuit, "circuit_terminations__location__in"),
+                             # Tuple of (Model, query_string)
+                             # For complex relationships
+        
+        (VirtualMachine, "cluster__location__in")
+                             # Another complex relationship example
+    ],
+    
+    # Inheritance from Base Panel
+    label="Statistics",
+    section=SectionChoices.RIGHT_HALF,
+    weight=700,
+    
+    # Template Configuration
+    body_content_template_path="components/panel/stats_panel_body.html"
+)
+```
+
+## Layouts and Sections
+
+### Section Choices
+- `LEFT_HALF`: Left side of the view
+- `RIGHT_HALF`: Right side of the view
+- `FULL_WIDTH`: Spans entire width
+
+### Panel Weights
+Standard panel weights for consistent ordering:
+```python
+WEIGHT_COMMENTS_PANEL = 200
+WEIGHT_CUSTOM_FIELDS_PANEL = 300
+WEIGHT_COMPUTED_FIELDS_PANEL = 400
+WEIGHT_RELATIONSHIPS_PANEL = 500
+WEIGHT_TAGS_PANEL = 600
+```
+
+//TODO: (include image) <img src="/api/placeholder/800/400" alt="Layout Sections" />
+*Layout sections and panel ordering*
+
+## Complete Example
+
+```python
+class LocationUIViewSet(NautobotViewSetMixin):
+    queryset = Location.objects.all()
+    
+    object_detail_content = ObjectDetailContent(
+        panels=(
+            ObjectFieldsPanel(
+                section=SectionChoices.LEFT_HALF,
+                fields=["name", "status", "type"],
+                weight=100,
+            ),
+            StatsPanel(
+                filter_name="location",
+                related_models=[
+                    Device,
+                    (Circuit, "circuit_terminations__location__in")
+                ],
+                section=SectionChoices.RIGHT_HALF,
+                weight=100,
+            ),
+            GroupedKeyValueTablePanel(
+                body_id="custom-fields",
+                data={
+                    "Network": {
+                        "VLAN": "100",
+                        "IP Range": "192.168.1.0/24"
+                    }
+                },
+                section=SectionChoices.FULL_WIDTH,
+                weight=300,
+            ),
+            ObjectTextPanel(
+                object_field="description",
+                render_as=BaseTextPanel.RenderOptions.MARKDOWN,
+                section=SectionChoices.FULL_WIDTH,
+                weight=400,
+            )
+        )
+    )
+```
+
+//TODO: (include image) <img src="/api/placeholder/800/600" alt="Complete Example" />
+*Complete example showing all panel types in action*
+
+## Best Practices
+
+### 1. Panel Organization
+- Use consistent weights across your application
+- Group related information in adjacent panels
+- Consider mobile viewports when choosing sections
+
+### 2. Performance
+- Be specific with field selections
+- Use appropriate indexes for StatsPanel queries
+- Consider lazy loading for heavy content
```

Comment:
This one could be a whole topic unto itself... it's not something we currently have any built-in functionality for though?